### PR TITLE
fix(system-restore): do not overwrite existing volume, PV, PVC

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/5064

### CODE REVIEW

File: controller/system_rollout_controller.go [hunk 1]
```diff
@@ -1222,10 +1222,8 @@ func (c *SystemRolloutController) restorePersistentVolumes() (err error) {
 			return err
 		}
 
-		restore.Spec.ClaimRef = exist.Spec.ClaimRef
 		if !reflect.DeepEqual(exist.Spec, restore.Spec) {
```
Comment (innobead) [inline controller/system_rollout_controller.go:1225]: There is an existing PV already, so it means we should not to restore it anyway no matter if the existing PV is the same as the restored PV or not. Then, why still need to check `if !reflect.DeepEqual(exist.Spec, restore.Spec)`? 

Also, it seems we will still annotate the PV with the remote backup URL and last restored time, but actually it's not restored. Do you think having another annotation to mark this resource has been processed but not restored. WDYT?


The same questions above should be applied to other resources like PVC and Volume here.

**Review by innobead — CHANGES_REQUESTED** [review]

File: controller/system_rollout_controller.go (hunk 1 above)
Comment (c3y1huang) [reply controller/system_rollout_controller.go]: > There is an existing PV already, so it means we should not to restore it anyway no matter if the existing PV is the same as the restored PV or not. Then, why still need to check `if !reflect.DeepEqual(exist.Spec, restore.Spec)`?

The intention is to output some logs. But it is not needed after introducing different annotations for the skipped resources. Removed.

>
> Also, it seems we will still annotate the PV with the remote backup URL and last restored time, but actually it's not restored. Do you think having another annotation to mark this resource has been processed but not restored. WDYT?

Fair. Introducing new annotations for the skipped resources. The annotations will replace the non-skipped annotations during the update.
- `longhorn.io/last-skipped-system-restored`
- `longhorn.io/last-skipped-system-restored-at`

Ex:
```
      longhorn.io/last-skipped-system-restored: s3://c3y1huang-s3@ap-southeast-1/backupstore/system-backu
ps/v1.5.0-dev/test1
      longhorn.io/last-skipped-system-restored-at: "2022-12-15T16:04:08Z"
```

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, just a few minor requests.

**Review by innobead — APPROVED** [review]
I think it's good to go :slightly_smiling_face:

**Comment by innobead** [comment]
@c3y1huang Please backport to 1.4.x. Thanks.

**Pull request merged**